index.js

```diff
@@ -18,27 +17,54 @@ function createOnResponseHandler(action) {
 }
 
 app.post('/', upload.single('thumb'), function (req, res, next) {
-  var payload = JSON.parse(req.body.payload);
+  const payload = JSON.parse(req.body.payload);
 
-  var validUser   = payload.Account.id === 1 || payload.Account.title === process.env.USERNAME;
-  var validPlayer = payload.Player.uuid === process.env.PLAYER;
+  const isValidUser = (
+    payload.Account.id === 1 ||
+    payload.Account.title === process.env.USERNAME
+  );
+
+  const isValidPlayer = payload.Player.uuid === process.env.PLAYER;
 
   // If the right player is playing a track, display a notification.
-  if (validPlayer && validUser && payload.Metadata.type === 'track') {
-    const common = `token=${process.env.TOKEN}`;
-    if (payload.event == "media.play" || payload.event == "media.resume") {
-      request.post({
-        headers: {'content-type' : 'application/x-www-form-urlencoded'},
-        url:     'https://slack.com/api/users.profile.set',
-        body:    `${common}&profile=%7B%22status_text%22%3A%22${encodeURIComponent(payload.Metadata.originalTitle || payload.Metadata.grandparentTitle)}%20-%20${encodeURIComponent(payload.Metadata.title)}%22%2C%22status_emoji%22%3A%22%3A${process.env.PLAY_EMOJI}%3A%22%7D`
-      }, createOnResponseHandler('play'));
-    } else if (payload.event == "media.pause" || payload.event == "media.stop") {
-      request.post({
-        headers: {'content-type' : 'application/x-www-form-urlencoded'},
-        url:     'https://slack.com/api/users.profile.set',
-        body:    `${common}&profile=%7B%22status_text%22%3A%22%22%2C%22status_emoji%22%3A%22%3A${process.env.PAUSE_EMOJI}%3A%22%7D`
-      }, createOnResponseHandler('stop'));
-    }
+  const isValidRequestType = (
+    isValidPlayer &&
+    isValidUser &&
+    payload.Metadata.type === 'track'
+  );
+
+  const isPlayEvent = isValidRequestType &&
+    (payload.event == 'media.play' || payload.event == 'media.resume');
+
+  const isPauseEvent = isValidRequestType &&
+    (payload.event == 'media.pause' || payload.event == 'media.stop');
+
+  if (isPlayEvent || isPauseEvent) {
+    const statusText = isPlayEvent ?
+      `${payload.Metadata.originalTitle || payload.Metadata.grandparentTitle} - ${payload.Metadata.title}` :
+      '';
+
+    const playEmoji = req.query.playEmoji || process.env.PLAY_EMOJI;
+    const pauseEmoji = req.query.pauseEmoji || process.env.PAUSE_EMOJI;
+    const token = process.env.TOKEN;
+
+    const params = [
+      ['token', token],
+      ['profile', JSON.stringify({
+        status_text: statusText,
+        status_emoji: `:${isPlayEvent ? playEmoji : pauseEmoji}:`
```

Comment:
Can we add the implicit `:`s only if the emoji is a string longer than 1 character? That would solve #4 at the same time.